presto-native-execution/presto_cpp/main/types/tests/RestFunctionHandleTest.cpp

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/types/PrestoToVeloxExpr.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/remote/client/Remote.h"
+#include "velox/functions/remote/server/RemoteFunctionService.h"
+
+using namespace facebook::presto;
+using namespace facebook::velox;
+
+class RestFunctionHandleTest : public ::testing::Test {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance(memory::MemoryManager::Options{});
+  }
+
+  void setupCallExpression() {
+    memoryPool_ = memory::MemoryManager::getInstance()->addLeafPool();
+    converter_ =
+        std::make_unique<VeloxExprConverter>(memoryPool_.get(), &typeParser_);
+
+    expectedMetadata_.serdeFormat = functions::remote::PageFormat::PRESTO_PAGE;
+
+    testExpr_.returnType = "bigint";
+    testExpr_.displayName = "testFunction";
+    auto cexpr = std::make_shared<protocol::ConstantExpression>();
+    cexpr->type = "bigint";
+    cexpr->valueBlock.data = "CgAAAExPTkdfQVJSQVkBAAAAAAEAAAAAAAAA";
+    testExpr_.arguments.push_back(cexpr);
+
+    auto cexpr2 = std::make_shared<protocol::ConstantExpression>();
+    cexpr2->type = "bigint";
+    cexpr2->valueBlock.data = "CgAAAExPTkdfQVJSQVkBAAAAAAEAAAAAAAAA";
+    testExpr_.arguments.push_back(cexpr2);
+  }
+
+  void SetUp() override {
+    auto restConfig = restSystemConfig();
+    auto systemConfig = SystemConfig::instance();
+    systemConfig->initialize(std::move(restConfig));
+    setupCallExpression();
+  }
+
+  static std::unique_ptr<config::ConfigBase> restSystemConfig(
+      const std::unordered_map<std::string, std::string>& configOverride = {}) {
+    std::unordered_map<std::string, std::string> systemConfig{
+        {std::string(SystemConfig::kRemoteFunctionServerSerde),
+         std::string("presto_page")},
+        {std::string(SystemConfig::kRemoteFunctionServerRestURL),
+         std::string("http://localhost:8080")}};
+
+    for (const auto& [configName, configValue] : configOverride) {
+      systemConfig[configName] = configValue;
+    }
+    return std::make_unique<config::ConfigBase>(std::move(systemConfig), true);
+  }
+
+  std::shared_ptr<protocol::RestFunctionHandle> functionHandle_;
+  protocol::CallExpression testExpr_;
+  functions::RemoteVectorFunctionMetadata expectedMetadata_;
+  std::shared_ptr<memory::MemoryPool> memoryPool_;
+  TypeParser typeParser_;
+  std::unique_ptr<VeloxExprConverter> converter_;
+};
+
+TEST_F(RestFunctionHandleTest, parseRestFunctionHandle) {
+  try {
```

Comment:
Add a test for a function signature with parameterizsed type like DECIMAL. Also all the complex types ARRAY, MAP, STRUCT (with nesting) if supported.